### PR TITLE
Sw 6932 deputy documents

### PR DESCRIPTION
**Pull request by kate-49**

### CODE REVIEW

File: internal/model/document.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package model
+
+type Document struct {
+	Id                  int    `json:"id"`
+	Type                string `json:"type"`
+	FriendlyDescription string `json:"friendlyDescription"`
+	CreatedDate         string `json:"createdDate"`
+	Direction           string `json:"direction"`
+	Filename            string `json:"filename"`
+	CreatedBy           User
```
Comment (denialanderror) [inline internal/model/document.go:10]: Missing json tag

File: internal/server/add_document.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+			Date:    time.Now().Format("2006-01-02"),
+		}
+
+		documentDirectionRefData, err := client.GetRefData(getContext(r), "/documentDirection")
+		if err != nil {
+			return err
+		}
+		vars.DocumentDirectionRefData = documentDirectionRefData
+
+		documentTypes, err := client.GetRefData(getContext(r), "?filter=noteType:deputy")
```
Comment (denialanderror) [inline internal/server/add_document.go:44]: If you are going to use ref data for this, you should at least make it async with goroutines. Otherwise to just load the page, the client has to fetch one set of refdata, wait for that to return, then send off a second request for the next lot.

File: internal/sirius/get_refdata.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/ministryofjustice/opg-sirius-supervision-deputy-hub/internal/model"
+	"io"
+	"net/http"
+	"strings"
+)
+
+func (c *Client) GetRefData(ctx Context, refDataType string) ([]model.RefData, error) {
```
Comment (denialanderror) [inline internal/sirius/get_refdata.go:12]: What's the reason for this change? By passing in refDataType as a string, you've lost all type safety and removed the interfaces indicating what refdata is fetchable.

If you are going down this route, it would be much better to make GetRefData private (lower case) and export the functions for fetching the specific refdatas, which then just call this one with string values.

**Review by denialanderror — CHANGES_REQUESTED** [review]

File: internal/sirius/get_refdata.go (hunk 3 above)
Comment (josephsmith0705) [reply internal/sirius/get_refdata.go:12]: The reason was to get rid of duplicated code across these refdata endpoints, but I hadn't thought about type safety - I'll give that a shot

**Review by denialanderror — APPROVED** [review]
